module/friendship/adapter/postgres/repository/subscription_test.go

```diff
@@ -0,0 +1,390 @@
+				Status:       domain.SubscriptionStatusSubscribed,
+			}
+			suite.prepareSubscription(t, ctx, sub)
+			createdSub := sub
+			if tc.err != nil {
+				if tc.modifySub.SubscriberID != "" {
+					createdSub.SubscriberID = "subscriber-id"
+				} else if tc.modifySub.UserID != "" {
+					createdSub.UserID = "user-id"
+				}
+			}
+			var err error
+			sub.Id, err = repo.Create(ctx, createdSub)
+			if tc.err == nil {
+				assert.NoError(t, err)
+				assert.NotEmpty(t, sub.Id)
+			} else {
+				assert.Error(t, err)
+				assert.Empty(t, sub.Id)
+			}
+
+			suite.rollbackSubscription(t, ctx, sub, []string{sub.Id})
+		})
+
+	}
+}
+
+func TestSubscription_UpdateStatus(t *testing.T) {
+	ctx := context.Background()
+	suite := NewSuite(ctx)
+	repo := NewSubscriptionRepository(suite.db)
+
+	errDB := errors.New("some error from db")
+
+	cs := []SubscriptionTestCase{
+		{
+			name: "successful",
+		},
+		{
+			name: "fail by invalid id",
+			modifySub: domain.Subscription{
+				Base: domain.Base{Id: "sub-id"},
+			},
+			err: common.ErrDB(errDB),
+		},
+	}
+	for _, tc := range cs {
+		t.Run(tc.name, func(t *testing.T) {
+			sub := domain.Subscription{
+				UserID:       util.GenUUID(),
+				SubscriberID: util.GenUUID(),
+				Status:       domain.SubscriptionStatusSubscribed,
+			}
+			suite.prepareSubscription(t, ctx, sub)
+			var err error
+			sub.Id, err = repo.Create(ctx, sub)
+			assert.NoError(t, err)
+			id := sub.Id
+			if tc.modifySub.Id != "" {
+				id = tc.modifySub.Id
+			}
+
+			repo.UpdateStatus(ctx, id, domain.SubscriptionStatusUnsubscribed)
+			result, err := repo.GetSubscription(ctx, domain.Subscriptions{sub})
+			assert.NoError(t, err)
+			if tc.err == nil {
+				assert.Equal(t, domain.SubscriptionStatusUnsubscribed, result[0].Status)
+			} else {
+				assert.Equal(t, domain.SubscriptionStatusSubscribed, result[0].Status)
+			}
+
+			suite.rollbackSubscription(t, ctx, sub, []string{sub.Id})
+		})
+	}
+}
+
+func TestSubscription_UnsertSubscription(t *testing.T) {
+	ctx := context.Background()
+	suite := NewSuite(ctx)
+	repo := NewSubscriptionRepository(suite.db)
+
+	cs := []SubscriptionTestCase{
+		{
+			name: "successful because not existing subscription",
+		},
+		{
+			name:      "successful because not existing subscription",
+			isExisted: true,
+		},
+	}
+	for _, tc := range cs {
+		t.Run(tc.name, func(t *testing.T) {
+			sub := domain.Subscription{
+				UserID:       util.GenUUID(),
+				SubscriberID: util.GenUUID(),
+				Status:       domain.SubscriptionStatusSubscribed,
+			}
+			suite.prepareSubscription(t, ctx, sub)
+			var err error
+			if tc.isExisted {
+				sub.Id, err = repo.Create(ctx, sub)
+				assert.NoError(t, err)
+			}
+			upsertSub := sub
+			upsertSub.Status = domain.SubscriptionStatusUnsubscribed
+			var upsertErr error
+			sub.Id, upsertErr = repo.UnsertSubscription(ctx, upsertSub)
+			assert.NoError(t, upsertErr)
+
+			result, err := repo.GetSubscription(ctx, domain.Subscriptions{sub})
```

Comment:
Should only test ONE function. That's why it's call `unit` test

Reply:
Here, I don't use `repo.GetSubscription` to test itself, I use it to check Is the `UpsertSubscription` function working good that is have change data.